app/cli/main.go

```diff
@@ -16,25 +16,86 @@
 package main
 
 import (
+	"crypto/sha256"
+	"fmt"
 	"os"
+	"strings"
+	"time"
 
 	"github.com/chainloop-dev/chainloop/app/cli/cmd"
+	"github.com/chainloop-dev/chainloop/app/cli/internal/telemetry"
+	"github.com/chainloop-dev/chainloop/app/cli/internal/telemetry/posthog"
 	v1 "github.com/chainloop-dev/chainloop/app/controlplane/api/controlplane/v1"
 	"github.com/go-kratos/kratos/v2/errors"
 	jwtMiddleware "github.com/go-kratos/kratos/v2/middleware/auth/jwt"
 	"github.com/rs/zerolog"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 	"google.golang.org/grpc/status"
 )
 
+var (
+	// Posthog API key and endpoint to be injected in the build process
+	posthogAPIKey   = ""
+	posthogEndpoint = ""
+)
+
 func main() {
 	// Couldn't find an easier way to disable the timestamp
 	logger := zerolog.New(zerolog.ConsoleWriter{Out: os.Stderr, FormatTimestamp: func(interface{}) string { return "" }})
 	rootCmd := cmd.NewRootCmd(logger)
 
-	if err := rootCmd.Execute(); err != nil {
-		msg, exitCode := errorInfo(err, logger)
+	// Set the exit code, in case of an error it will be updated
+	var exitCode int
+
+	// Start the timer to measure the execution time
+	timeStart := time.Now()
+	// Run the command and get the executed command
+	executedCmd, err := rootCmd.ExecuteC()
+	if err != nil {
+		var msg string
+		msg, exitCode = errorInfo(err, logger)
 		logger.Error().Msg(msg)
-		os.Exit(exitCode)
+	}
+	// Measure the execution time
+	executionTime := time.Since(timeStart)
+
+	if !cmd.IsTelemetryDisabled() {
+		recordCommand(executedCmd, exitCode, executionTime)
+	}
+
+	os.Exit(exitCode)
+}
+
+// recordCommand sends the command to the telemetry service
+func recordCommand(executedCmd *cobra.Command, exitCode int, executionTime time.Duration) {
+	telemetryClient, err := posthog.NewClient(posthogAPIKey, posthogEndpoint)
+	if err != nil {
+		return
+	}
+
+	cmdTracker := telemetry.NewCommandTracker(telemetryClient)
+	tags := telemetry.Tags{
+		"cli_version":       cmd.Version,
+		"exit_code":         fmt.Sprint(exitCode),
+		"cp_url_hash":       hashControlPlaneURL(),
+		"chainloop_source":  "cli",
+		"execution_time_ms": fmt.Sprint(executionTime.Milliseconds()),
+	}
+
+	// It tries to extract the token from the context and add it to the tags. If it fails, it will ignore it.
+	contextToken := executedCmd.Context().Value(cmd.AuthenticationToken{})
+	if contextToken != nil {
+		parsedToken, ok := contextToken.(*cmd.ParsedToken)
+		if ok {
+			tags["token_type"] = parsedToken.Type
```

Comment:
I get a panic when using locally a CHAINLOOP_API_TOKEN during an attestation

```
./bin/chainloop att init --replace
ERR rpc error: code = InvalidArgument desc = when using an API Token, workflow name is required as parameter
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x201ea00]

goroutine 1 [running]:
main.recordCommand(0xc000f47508, 0x1, 0x18fffa5e)
        /home/migmartri/work/chainloop/chainloop/app/cli/main.go:91 +0x3a0
main.main()
        /home/migmartri/work/chainloop/chainloop/app/cli/main.go:64 +0x2c8
```

Reply:
Ok, the problem was that my token didn't have `org_id` since the current live backend doesn't have it yet. 

but I think this is surfacing another problem, that the `parseToken` method can return 3 values, `nil`, empty token `&ParsedToken{}` or a full-fledged token. 

We should probably only return `nil` or a token and handle it accordingly.